File: AppTool/src/main/java/com/luoye/apptool/lazy/LazyFragment.java
package com.luoye.apptool.lazy;

import androidx.fragment.app.Fragment;

import com.luoye.apptool.lazy.LazyInterFace;

/**
 * Created by: luoye
 * Time: 2022/12/23
 * user: 懒加载
 */
public abstract class LazyFragment extends Fragment {

    protected String TAG = "---BaseFragment";

    public LazyFragment() {
        if (this instanceof LazyInterFace) {
            lazyInterFace = (LazyInterFace) this;
        }
    }

    private boolean isLoaded = false;

    private LazyInterFace lazyInterFace;

    private void lazyInit() {
        if (lazyInterFace != null) lazyInterFace.lazyInit();
    }


    @Override
    public void onResume() {
        super.onResume();
        if (!isLoaded && !isHidden()) {
            lazyInit();
            isLoaded = true;
        }
    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        isLoaded = false;
    }

}

File: ktBase/src/main/java/com/luoye/ktbase/lazy/LazyFragment.kt
package com.luoye.ktbase.lazy

import androidx.fragment.app.Fragment

/**
 *
 * Created by: luoye
 * Time: 2022/12/23
 * user: 懒加载
 *
 */
abstract class LazyFragment() : Fragment() {

    protected var TAG = "---BaseFragment"

    private var isLoaded = false

    private var lazyInterFace: LazyInterFace? = null

    init {
        if (this is LazyInterFace) {
            lazyInterFace = this
        }
    }

    private fun lazyInit() {
        lazyInterFace?.lazyInit()
    }


    override fun onResume() {
        super.onResume()
        if (!isLoaded && !isHidden) {
            lazyInit()
            isLoaded = true
        }
    }


    override fun onDestroyView() {
        super.onDestroyView()
        isLoaded = false
    }


}